state_machine/plugs/flight/lib/pycubed.py

```diff
@@ -220,216 +317,339 @@ def RGB(self, value):
+        1. NVM registers(boot count, flags, counters)
+        2. Time (seconds) since boot/hard reset
+        3. Battery voltage as % of full
+        """
+
         self._stat.update({
             'boot-time': self.BOOTTIME,
             'boot-count': self.c_boot,
             'time-on': self.timeon,
             'fuel-gauge': self.fuel_gauge,
             'flags': {
-                'deploy': self.f_deploy,
-                'mid-deploy': self.f_mdeploy,
-                'burn1': self.f_burn1,
-                'burn2': self.f_burn2
-            }, 'counters': {
-                'state-errors': self.c_state_err,
-                'vbus-resets': self.c_vbus_rst,
-                'deploy': self.c_deploy,
-                'downlink': self.c_downlink,
-            },
+                    'deploy': self.f_deploy,
+                    'mid-deploy': self.f_mdeploy,
+                    'burn1': self.f_burn1,
+                    'burn2': self.f_burn2
+                    },
+            'counters': {
+                    'state-errors': self.c_state_err,
+                    'vbus-resets': self.c_vbus_rst,
+                    'deploy': self.c_deploy,
+                    'downlink': self.c_downlink,
+                    },
         })
+
         self._stat.update({
-            'raw': bytes([self.micro.nvm[_BOOTCNT],
-                          self.micro.nvm[_FLAG],
-                          self.micro.nvm[_RSTERRS],
-                          self.micro.nvm[_DWNLINK],
-                          self.micro.nvm[_DCOUNT]]) +
-            self.BOOTTIME.to_bytes(3, 'big') +
-            self._stat['time-on'].to_bytes(4, 'big') +
-            int(self._stat['fuel-gauge']).to_bytes(1, 'big')
+            'raw': bytes
+            ([
+                self.micro.nvm[_BOOTCNT],
+                self.micro.nvm[_FLAG],
+                self.micro.nvm[_RSTERRS],
+                self.micro.nvm[_DWNLINK],
+                self.micro.nvm[_DCOUNT]
+            ])
+            + self.BOOTTIME.to_bytes(3, 'big')
+            + self._stat['time-on'].to_bytes(4, 'big')
+            + int(self._stat['fuel-gauge']).to_bytes(1, 'big')
         })
+
         return self._stat
 
     @property
     def timeon(self):
+        """
+        return the time on a monotonic clock
+        """
         return int(time.monotonic())
 
     def crc(self, data):
+        """
+        cyclic redundancy check (crc)
+        """
         crc = 0
+
+        # hash function: xor each byte with current crc and return
         for byte in data:
             crc ^= byte
+
         return crc
 
     def new_file(self, substring, binary=False):
-        '''
-        substring something like '/data/DATA_'
-        directory is created on the SD!
-        int padded with zeros will be appended to the last found file
-        '''
+        """
+        create a new file on the SD card
+        substring example: '/data/DATA_'
+        int padded with zeroes will be appended to the last found file
```

Comment:
This is a little confusing to me

Reply:
Since this is mostly being carried over from the previous file, let's leave it alone. I agree that this function is a bit odd, but I think we should resolve that later.